### PR TITLE
Fixing checkstyle overlap

### PR DESCRIPTION
**Pull request by ctripoll**
This pull request fixes the checkstyle overlap that was affecting the `/deals` page and the `/shoppingcart` page. Each scss sheet is independent now, with unique naming conventions.

Additionally, I have made changes to `/shoppingcart` from suggestions/feedback received from Shraddha on 4/25.

Lastly, merging in `/history`.

### CODE REVIEW

**Comment by cyan33** [comment]
Does this have anything to do with the `menu`? Because I'm working on it right now and will likely submit a PR very soon. And this PR still contains some duplicated `menu` code.

Do you mind merging mine first?

**Comment by ctripoll** [comment]
@cyan33  Hey I merged your menu changes and resolved the conflicts here

File: src/components/History.js [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+@inject('historyStore')
+@observer
+export default class History extends Component {
+
+	  state = {
+			  isLoadingHistory: true,
+			    history: null
+  }
+	  
+	  constructor(props){
```
Comment (cyan33) [inline src/components/History.js:19]: This should be removed. Also the `toggleShow`, `hide` methods.

File: src/components/History.js [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+	  }
+
+  render() {
+	  const { history } = this.state
+	    return (
+	    
+	      <div className="baskets-container">
+	        <div className="top-logo" style={{zIndex:999}} >
+	        	<button className="btn-primary" type="button" onClick={this.toggleShow} onBlur={this.hide}>
+	        	<img src={logo} alt="logo" style={{zIndex:999}} />
```
Comment (cyan33) [inline src/components/History.js:102]: This part of the code is extracted into `<Menu />`

File: src/components/ShoppingCart.js [hunk 3]
```diff
@@ -71,22 +67,28 @@ export default class ShoppingCart extends Component {
...
+          <div className="basket-header">
+    	      <label>Grocery Basket</label>
+	        </div>
 					{
 						deals &&
 						<Fragment>
 							<ul className="deals-list">
 								{this.getDealItems()}
 							</ul>
+              <div className="basket-total-purchase">Total Purchase Cost: $57.39</div>
```
Comment (cyan33) [inline src/components/ShoppingCart.js:83]: Can this value be derived from calculation instead?

**Comment by cyan33** [comment]
Good stuff! I've made some comments. About the `<Menu />` refactor, you can refer to [my previous PR](). The top-logo and the menu items are located in the `<Menu />` Component now and you could just put it there. Thanks

**Comment by ctripoll** [comment]
@cyan33  For the first two in `History.js`, do you mind making those changes? I think I understand but I would hate to mess it up as it is going to master.

For the calculation in `/ShoppingCart.js`. It can be a calculation, I did it very quickly last night and made a quick placeholder.

**Comment by cyan33** [comment]
Cool I'll merge it then. Good work!

**Pull request merged**